SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialAuthorityTest/SpatialAuthorityTest.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "SpatialFunctionalTest.h"
+#include "SpatialAuthorityTest.generated.h"
+
+class ASpatialAuthorityTestActor;
+class ASpatialAuthorityTestReplicatedActor;
+
+/** Check SpatialAuthorityTest.cpp for Test explanation. */
+UCLASS()
+class SPATIALGDKFUNCTIONALTESTS_API ASpatialAuthorityTest : public ASpatialFunctionalTest
+{
+	GENERATED_BODY()
+
+public:
+	ASpatialAuthorityTest();
+
+	virtual void BeginPlay() override;
+
+	virtual void GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const override;
+
+	virtual void FinishStep() override { ResetTimer(); Super::FinishStep();	};
```

Comment:
Does this also get called on workers which do not have the step assigned to them? E.g. Will Step(Server(1)) also reset the timer for Server2?

Reply:
No, this is a local timer. On BeginPlay() they reset it, then every time they finish a step they also reset it (so it's ready for the next step). This was for avoiding having to add StartEvent on them just to set the timer, so the idea is the previous step always leaves the local timer in a good state.